### DISCUSSION TITLE
style bw, lexer psql rendering as plaintext via minted

### DISCUSSION THREAD

**Discussion by goyalyashpal** [pygments/pygments#2556] [Q&A] [answered]
* the `style=bw`  (used via minted package in latex) is rendering for the lexer `psql` to be just like plain text (the default style is working fine though)
* whereas i would have assumed that it would make:
    * the keywords as bold and 
    * the `out_token` output part as light gray (i.e. no change from the default style).

i have seen the visual sample renderings given at the page https://pygments.org/styles/ and it also shows the keywords to be bold in bw style, 

so, why it ain't working with psql lexer & how can i make it work?

i also want to make the `out_token` to have a different theme (dark) if possible - so as to give it look of an actual screenshot of a terminal

----
Edit:
tried at CLI with `-O style=bw`, and keywords are correctly bolden there, the output_tokens are still plaintext though
but regardless, as for the keywords, maybe the problem is at the minted part 🤔 

`pygmentize -l=psql -O=style=bw -F=keywordcase:case=upper test.mysqlcon`

the formatter is [latex] i guess

[latex]: https://pygments.org/docs/formatters/#LatexFormatter

----
Source files & screenshots:

i had already modified the prompt from `mysql>` to `mysql->` to match the <tt>[re_prompt]</tt> pattern.

[re_prompt]: https://github.com/pygments/pygments/blob/0fba7b8a1dfc905e71c70ae7c29b78ce5baed3e8/pygments/lexers/sql.py#L270 "pygments/lexers/sql.py#L270"

<details open><summary>source file contents <i>(click to expand)</i></summary>

<br/>

```mysql-session
mysql-> create database custom_sample;
Query OK, 1 row affected (0.04 sec)

mysql-> drop database custom_sample;
Query OK, 0 rows affected (0.05 sec)

mysql-> create table if not exists std (
    ->     id int,
    ->     name varchar(15),
    ->     gender char
    -> );
Query OK, 0 rows affected (0.03 sec)

mysql-> show tables from meradb;
+------------------+
| Tables_in_meradb |
+------------------+
| std              |
+------------------+
1 row in set (0.00 sec)

```

</details> 

<details open><summary>screenshots: </summary>

* cli <br/>
![image](https://github.com/pygments/pygments/assets/19423063/8176873b-f664-4cd7-a066-ae552678fc7c)

* pdf <br/>
![image](https://github.com/pygments/pygments/assets/19423063/5b766cb7-0b74-48c7-b2c3-8dbb95a184bb)

</details>

**Comment by birkenfeld** [comment]
Can you share `test.mysqlcon`?

**Reply by goyalyashpal** [reply]
yeah sure:

i have modified the prompt from `mysql>` to `mysql->` to match the <tt>[re_prompt]</tt> pattern.

[re_prompt]: https://github.com/pygments/pygments/blob/0fba7b8a1dfc905e71c70ae7c29b78ce5baed3e8/pygments/lexers/sql.py#L270 "pygments/lexers/sql.py#L270"

<details><summary>source file contents <i>(click to expand)</i></summary>

<br/>

```mysql-session
mysql-> create database custom_sample;
Query OK, 1 row affected (0.04 sec)

mysql-> drop database custom_sample;
Query OK, 0 rows affected (0.05 sec)

mysql-> create table if not exists std (
    ->     id int,
    ->     name varchar(15),
    ->     gender char
    -> );
Query OK, 0 rows affected (0.03 sec)

mysql-> show tables from meradb;
+------------------+
| Tables_in_meradb |
+------------------+
| std              |
+------------------+
1 row in set (0.00 sec)

```

</details> 


<details><summary>screenshots: </summary>

* cli <br/>
![image](https://github.com/pygments/pygments/assets/19423063/8176873b-f664-4cd7-a066-ae552678fc7c)

* pdf <br/>
![image](https://github.com/pygments/pygments/assets/19423063/5b766cb7-0b74-48c7-b2c3-8dbb95a184bb)

</details>

**Comment by goyalyashpal** [comment, marked as answer]
so, it was a font issue with `pdflatex`; i switched to `xelatex` and a very slightly bold font appeared by default. 
the switch was _extremely_ easy as the binary is included in miktex distribution already, and texstudio has the build commands covered. so, it was a matter of _just_ invoking the predefined xelatex command.

<img src="https://github.com/pygments/pygments/assets/19423063/d31fd98d-6e8c-4f3f-b5ac-f694860e3ecf" width=400px/>

<br/>

see the TEX.SE answer quoted below (emphasis mine)

> Another customization that is much easier is to set the font used by minted. This is the easiest with _XeLaTeX_ or _LuaLaTeX_ where you can set the monospaced font (i.e., the font used for minted, and also for \texttt, \verb etc) with the fontspec package. For _pdfLaTeX_ there are various packages to set the document font (or sometimes only the mono font) but there the choices are much more limited. 
>
> The choice of font is important because _not all monospaced fonts support boldface_, for example the default font for minted does not have bold _so the normal thickness will be displayed even if the style file specifies bold_.
> 
> \- @Marijn at https://tex.stackexchange.com/q/584071/584212#584212 "Configure minted style in LaTeX for code highlighting"
